app/src/main/java/at/bitfire/icsdroid/db/dao/CredentialsDao.kt

```diff
@@ -0,0 +1,59 @@
+package at.bitfire.icsdroid.db.dao
+
+import android.content.Context
+import android.database.SQLException
+import android.database.sqlite.SQLiteConstraintException
+import androidx.annotation.WorkerThread
+import androidx.room.Dao
+import androidx.room.Query
+import at.bitfire.icsdroid.db.AppDatabase
+import at.bitfire.icsdroid.db.entity.Credential
+import at.bitfire.icsdroid.db.entity.Subscription
+
+/**
+ * Creates an interface with all the credentials stored in the database.
+ * @see AppDatabase
+ * @see Credential
+ */
+@Dao
+interface CredentialsDao {
+    companion object {
+        /**
+         * Alias for [AppDatabase.getInstance] -> [AppDatabase.credentialsDao].
+         * @param context The context that is requesting access to the dao.
+         */
+        fun getInstance(context: Context) = AppDatabase.getInstance(context).credentialsDao()
+    }
```

Comment:
I'd make the `AppDatabase.getInstance().xxxDao()` call explicit so no shortcut needed. (Same for Subscription)

Reply:
bffd744f4c3d169071da7dae09f9a73899e97aae